Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1338,9 +1339,11 @@ namespace Babylon
                 {
                     auto argumentObject{info[0].As<Napi::Object>()};
 
-                    XRRigidTransform* transform{XRRigidTransform::Unwrap(argumentObject)};
-                    if (transform != nullptr)
+                    const auto XRRigidTransformFunction = Env().Global().Get(XRRigidTransform::JS_CLASS_NAME).As<Napi::Function>();
+
+                    if (argumentObject.InstanceOf(XRRigidTransformFunction))
```

Comment:
Under what circumstances will this not be true?

Reply:
InstanceOf is a very slow check. Is there some property you can check instead?

Reply:
Using property check instead.